client/components/command-palette/use-command-filter.tsx

```diff
@@ -0,0 +1,22 @@
+export const MAGIC_SEPARATOR = '||';
+
+export const useCommandFilter = () => {
+	const commandFilter = ( value: string, search: string ) => {
+		const [ beforeSeparator, afterSeparator ] = value.split( MAGIC_SEPARATOR );
+
+		// Check if the search matches the part before the separator
+		if ( beforeSeparator.includes( search ) ) {
+			return 1;
+		}
+
+		// Check if there is an afterSeparator and the search matches it
+		if ( afterSeparator && afterSeparator.includes( search ) ) {
+			return 0.5;
+		}
+
+		// No match
+		return 0;
+	};
+
+	return [ commandFilter ];
```

Comment:
Is there any specific reason to return an array of one object?

If we decide to return more values, we can transform it to an array or object later.

```suggestion
	return commandFilter;
```

Reply:
Adjusted in https://github.com/Automattic/wp-calypso/pull/85418/commits/1d050721b5bb3d2449a150c5ef5964bbba1d7a6a